s2s_pipeline.py

```diff
@@ -330,14 +321,69 @@ def overwrite_device_argument(common_device: Optional[str], *handler_kwargs):
     else:
         raise ValueError("The TTS should be either parler, melo or chatTTS")
 
-    # 4. Run the pipeline
+
+def main():
+    (
+        module_kwargs,
+        socket_receiver_kwargs,
+        socket_sender_kwargs,
+        vad_handler_kwargs,
+        whisper_stt_handler_kwargs,
+        paraformer_stt_handler_kwargs,
+        language_model_handler_kwargs,
+        mlx_language_model_handler_kwargs,
+        parler_tts_handler_kwargs,
+        melo_tts_handler_kwargs,
+        chat_tts_handler_kwargs,
+    ) = parse_arguments()
+
+    setup_logger(module_kwargs.log_level)
+
+    optimal_mac_settings(
```

Comment:
Just to go a bit further in the refactoring, do you think having `optimal_mac_settings`, `check_mac_setttings` and `overwrite_device_argument` in prepare_all_args would improve clarity ?

Reply:
Not really, maybe the name for `prepare_args` is wrong, but the function has a very clear goal and adding these things would only make it more convoluted. We could rename `prepare_args` to something like -> `rename_args` and then have another method `prepare_args` where these methods also took part. WDYT?

Reply:
Yep indeed it is better this way! Let's have `prepare_args` renamed `rename_args` (I should have named it this way from the beginning), `optimal_mac_settings` and `if platform == "darwin":` part in a dedicated `prepare_module_args` and calls to `rename_args` and `overwrite_device_arguments` in `prepare_args` ?

Reply:
done!